File: src/lightnap-ng/src/app/core/backend-api/dtos/users/response/notification-search-results-dto.ts
import { PagedResponseDto } from "../../paged-response-dto";
import { NotificationDto } from "./notification-dto";

/**
 * Represents a paged response of notifications.
 * @extends PagedResponseDto<NotificationDto>
 * @property {number} unreadCount - The number of unread notifications.
 */
export interface NotificationSearchResultsDto extends PagedResponseDto<NotificationDto> {
  unreadCount: number;
}

File: src/LightNap.Core/Notifications/Dto/Response/NotificationSearchResultsDto.cs
using LightNap.Core.Api;

namespace LightNap.Core.Notifications.Dto.Response
{
    /// <summary>
    /// Contains the search results for notifications.
    /// </summary>
    public class NotificationSearchResultsDto(IList<NotificationDto> data, int pageNumber, int pageSize, int totalResults)
        : PagedResponseDto<NotificationDto>(data, pageNumber, pageSize, totalResults)
    {
        /// <summary>
        /// The number of unread notifications.
        /// </summary>
        public int UnreadCount { get; set; }
    }
}